dvc/project.py

```diff
@@ -154,6 +154,25 @@ def _check_circular_dependency(self, deps, outs):
         if circular_dependencies:
             raise CircularDependencyError(circular_dependencies.pop())
 
+    def _check_argument_duplication(self, outs, outs_no_cache, deps):
```

Comment:
Please add a separate function to avoid duplicating the same logic three times. Also it would be great if the exception that you raise had a hint about what argument was specified more than 1 time(e.g. was it a dependecy or output).